doc/articles/contributing/guidelines/creating-tests.md

```diff
@@ -0,0 +1,69 @@
+
+ Test type                   | Location
+ --------------------------- | ---------------------------------------------------------------------------------
+ UI tests                    | https://github.com/unoplatform/uno/tree/master/src/SamplesApp/SamplesApp.UITests
+ .NET Framework unit tests   | https://github.com/unoplatform/uno/tree/master/src/Uno.UI.Tests
+ Platform runtime unit tests | https://github.com/unoplatform/uno/tree/master/src/Uno.UI.RuntimeTests
+ Xaml code generation tests  | https://github.com/unoplatform/uno/tree/master/src/SourceGenerators/XamlGenerationTests
+
+ All these tests are run on each CI build, and all tests must pass before a PR can be merged.
+
+ ### UI tests
+
+ Uno's UI tests use the [Uno.UITest](https://github.com/unoplatform/Uno.UITest) testing harness, which mimics the [`Xamarin.UITest` API](https://docs.microsoft.com/en-us/appcenter/test-cloud/uitest/) and extends it to WebAssembly. These tests run out-of-process and interact with a running app (the [SamplesApp](https://github.com/unoplatform/uno/tree/master/src/SamplesApp/UITests.Shared)) to confirm that it behaves correctly on each supported platform.
+
+UI tests can mimic the actions of a user:
+
+ - tapping buttons and other UI elements
+ - entering keyboard input
+ - scrolling, swiping, and other gestures
+
+ They can verify assertions about the state of the app:
+
+  - Text labels
+  - any DependencyProperty value
+  - onscreen bounds of a view element
+  - comparing screenshots at different stages of the app, and asserting equality or inequality
+
+A complete set of instructions for authoring UI tests is available [here](../../uno-development/working-with-the-samples-apps.md).
+
+### .NET Framework unit tests (`Uno.UI.Tests`)
+
+These are 'classic' unit tests which are built against a mocked version of the `Uno.UI` assembly targetting .NET Framework. 
+
+These tests are ideal for testing platform-agnostic parts of the code, such as the dependency property system, or panel measurement logic. They [have access to](https://github.com/unoplatform/uno/blob/af5365331d87a80ed4dd83b9e4839cc0d4a0ee5b/src/Uno.UI/AssemblyInfo.cs#L3) internal Uno.UI members, and the mocking layer allows values to be read or written which can't be accessed via the 'real' API.
+
+### Platform runtime unit tests (`Uno.UI.RuntimeTests`)
+
+Again these are 'classic' unit tests, but they are run 'in-process' on the actual target platform, using the 'real' Uno.UI assemblies. They can be run locally through the [Unit Tests Runner](https://github.com/unoplatform/uno/blob/master/src/SamplesApp/SamplesApp.Shared/Samples/UnitTests/UnitTestsPage.xaml) sample in the SamplesApp.
+
+These tests are useful for testing behavior which can run synchronously and where correctness can be asserted programmatically. Relative to the .NET Framework tests, they have the advantage that platform-dependent behavior can be tested, but the disadvantage of being restricted to the public API surface.
```

Comment:
That's true, so they can. 

Though I would argue that where possible they _shouldn't_, since one of the advantages of the RuntimeTests is that they can easily be run on UWP. I forgot to mention that, actually.